windows-agent/cmd/ubuntu-pro-agent/agent/agent.go

```diff
@@ -229,3 +201,50 @@ func (a App) RootCmd() cobra.Command {
 func (a *App) SetArgs(args ...string) {
 	a.rootCmd.SetArgs(args)
 }
+
+// PublicDir creates a directory to store public data in.
+func (a *App) PublicDir() (string, error) {
+	homeDir := os.Getenv("UserProfile")
+	if homeDir == "" {
+		return "", errors.New("could not create public dir: %UserProfile% is not set")
+	}
+
+	publicDir := filepath.Join(homeDir, common.UserProfileDir)
+
+	if err := os.MkdirAll(publicDir, 0600); err != nil {
+		return "", fmt.Errorf("could not create public dir %s: %v", publicDir, err)
+	}
+
+	return publicDir, nil
+}
+
+// publicDir is a wrapper around PublicDir to allow overriding its path with an option.
+func (a *App) publicDir(opts options) (string, error) {
+	if opts.publicDir == "" {
+		return a.PublicDir()
+	}
+
+	if err := os.MkdirAll(opts.publicDir, 0600); err != nil {
```

Comment:
I am a bit uneasy that both `PublicDir()` and `publicDir()` create the directory. Can we maybe simplify in one of those which takes an option?

Reply:
I just didn't want the public interface to take this optional argument as it is a bit redundant. But I guess it's fine.

Reply:
In the end I inverted the logic (PublicDir calls publicDir).

I just did not want the App interface to have method `PublicDir(string)`. Feels like exposing its internals a bit.